src/site/content/en/blog/access-devices/index.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Accessing hardware devices on the web
+subhead: |
+  Pick the appropriate API to communicate with a hardware device of your choice.
+description: |
+  Pick the appropriate API to communicate with a hardware device of your choice.
+authors:
+  - beaufortfrancois
+date: 2021-01-15
+updated: 2021-01-15
+hero: hero.jpg
+thumbnail: thumbnail.jpg
+alt: A woman sitting in front of a wooden desk photo.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - capabilities
+  - devices
+---
+
+My goal is to help you, web developer, pick the appropriate API to communicate
+with a hardware device of your choice (e.g. a webcam, a microphone, a mouse, a
+keyboard, a stylus, an NFC tag, a Bluetooth device, or an exotic hardware
+device).
+
+There are many APIs to choose from. By answering a few questions on this device, you
+should find out the API to use and have some documentation to get started
+accessing this device.
+
+## 1. Do you want to access video stream from this device? {: #video }
+
+If so, use [MediaDevices.getUserMedia()] to get a live video stream from this
+device and learn about [capturing video]. You can also [control camera pan,
+tilt, zoom], and other camera settings such as [brightness & contrast], and even
+[take still images].
+
+## 2. Do you want to access audio stream from this device? {: #audio }
+
+Then, use [MediaDevices.getUserMedia()] to get a live audio stream from this
+device and learn about [capturing audio]. You can also use [Web Audio] to add
+effects to audio, create audio visualizations, apply spatial effects (such as
+panning). Check out [how to profile the performance of Web Audio apps] in Chrome
+as well.
+
+## 3. Do you want to receive input events from this device? {: #input }
+
+If so, try listening to [Keyboard] and [Pointer] events. If this device is a game
+controller, use the [Gamepad] API to know which buttons and axes are being
+pressed.
+
+## 4. Is this device an NFC tag? {: #nfc }
+
+Then, try [Web NFC] to read and write to this device when it's in close
+proximity to the user's device (usually 5-10 cm, 2-4 inches).
+
+## 5. Is this a Bluetooth device? {: #bluetooth }
+
+Click the button below and see if you can spot a nearby Bluetooth device that
+may be of interest to you. Note that this device may need to be in a special
+mode to be visible.
+
+<div class="w-text--center">
+<button class="w-button w-button--secondary w-button--with-icon"
+data-icon="bluetooth" onclick="navigator?.bluetooth?.requestDevice({ acceptAllDevices:true })
+|| alert('This browser does not support Web Bluetooth. Try Chrome.')">
+Request bluetooth device
+</button>
```

Comment:
I love it! I wonder if it will survive the web.dev team's review.

Reply:
:D

Reply:
Where I am there are so many devices in the local environment it is a little hard to interpret the results. I would point developers at a larger article that explains how to use chrome://bluetooth-internals and/or the nRF Connect app for Bluetooth discovery and reverse engineering.

Reply:
In general I prefer for these to be part of an iframe embed (glitch or codepen).